tensordict/tensordict.py

```diff
@@ -1515,6 +1515,8 @@ def unflatten_keys(
                 tensordict[new_key] = value
                 if inplace:
                     del self[old_key]
+                if key in self:
+                    tensordict.update(self[key])
```

Comment:
I don't think this needs to be in the inner for-loop right? Could be done immediately after instantiating `tensordict`.

Reply:
Yeah. Just moved it to the outer loop.